app/src/main/java/com/zulip/android/activities/ZulipActivity.java

```diff
@@ -1113,11 +1113,17 @@ private void sendMessage() {
                 return;
             } else {
                 try {
-                    Cursor streamCursor = makeStreamCursor(streamActv.getText().toString());
-                    if (streamCursor.getCount() == 0) {
+                    //check whether stream exists or not
+                    if (!isStreamExist(streamActv.getText().toString())) {
```

Comment:
https://github.com/zulip/zulip-android/blob/master/app/src/main/java/com/zulip/android/models/Stream.java#L162

Has a method getByName but this method creates a stream if not exists, so create a method in this Stream.java naming appropriately and this method returns null then not exists and if it does used the subscribed boolean in this stream to check if we are subscribed!  

_[Not related to this issue]_ We should consider to reduce the code in the ZulipActivity.java file and maybe split the codes wherever possible its 1899 lines currently!